src/elli/result.py

```diff
@@ -453,17 +456,20 @@ def as_delta_range(self, lower: int, upper: int):
 class ResultList:
     """Class to make a row of Results easier to handle."""
 
-    def __init__(self, results: List[Result] = None) -> None:
+    def __init__(self, results: List[Result] = None, mean: bool = False) -> None:
```

Comment:
Wouldn't it be better to follow the functional style numpy and others are using, i.e., something like

```python
res_list = ResultList([my, fancy, results])
res_list.mean() # or .mean w/o ()
```

that way users are able to choose whether they want to see the whole or the mean data _after_ they created the object.

Reply:
That would be better, but it complicates things. I will have a look if i can make it work.

Reply:
Something like this?

```python
def mean():
   return {'r_sp': np.mean(
                 np.squeeze(
                     np.array([getattr(result, 'r_sp') for result in self.results])
                 ),
                 axis=0,
             ), ....}
```

I think there is also a function to get all available attributes. That way the dict can be constructed generically.
Could also be included in the `__gettattr__` of the `ResultList` which would avoid the `mean` (I think this is the common way in python and pandas).

But I think it's also fine the way you did it. I will approve and if you decide to change it feel free to change it and re-request review or merge directly.

Reply:
This would work with the `resultlist.mean.psi` notation. It is short, but janky in its own way :wink: 
Whats nice is, that you could create the averagedList object and use it directly if you would want that.

```python
def __getattr__(self, name: str) -> npt.NDArray:
        if name == "mean":
            return AveragedResultList(self.results)

        return np.squeeze(np.array([getattr(result, name) for result in self.results]))


class AveragedResultList(ResultList):
    def __getattr__(self, name: str) -> npt.NDArray:
        return np.mean(super().__getattr__(name), axis=0)
```

Reply:
It's nice. I like it